examples/high_level_functions_demos/imperfections_demo.js

```diff
@@ -0,0 +1,194 @@
+var member = new Member();
+member.Beam(undefined, [node1.no, node2.no], section.GetNo());
+
+include("../includes/Tools/high_level_functions_support.js");
+var nodesForMembers = createNodesGrid(1, 8, [8, 1], [3, 0]);
+
+var member2 = new Member();
+var members_list = [];
+
+for (var i = 0; i < nodesForMembers.length - 1; ++i) {
+    members_list.push(member2.Beam(undefined, [nodesForMembers[i], nodesForMembers[i + 1]], section.GetNo()));
+}
+
+var memberSet = new MemberSet();
+memberSet.ContinuousMembers(undefined, members_list);
+
+if (RFEM) {
+    var thickness = new Thickness();
+    thickness.Uniform(1, "Thickness", material.GetNo(), 0.250);
+
+    var nodesForSurfaces = createNodesGrid(1, 10, [6, 2], [3, 3]);
+    dictSurfaces = createSurfacesFromNodesGrid(nodesForSurfaces, [3, 1], surfaces.TYPE_STANDARD, thickness.GetNo());
+    var surfaceList = Object.keys(dictSurfaces).map(function(key){
+        return dictSurfaces[key][0];
+    });
+
+    var surfaceSet = new SurfaceSet();
+    surfaceSet.ContinuousSurfaces(undefined, [surfaceList[2].no]);   
+}
+
+/*********************************************************************************************
+****************************************** Main **********************************************
+*********************************************************************************************/
+
+/****************************************************** Imperfection cases ***************************************************************/
+var imperfection_case = new ImperfectionCase();
+imperfection_case.LocalImperfection(undefined, [lc.GetNo()], undefined, true, "Imperfection case comment");
+var localIC = imperfection_case.GetNo();
+imperfection_case.Notional(undefined, [lc2.GetNo()], lc4.GetNo(), undefined, true, "Imperfection case Notional comment");
+// With global coordination system
+imperfection_case.InitialSway(undefined, [lc3.GetNo()], [[1.0], [2.0, 2.5, -2.6, "Only one comment"], [3.0, 1.5, -1.5]], 1, "GLOBAL_IN_X", "DIRECTION_YZ");
+// With user defined coordination system
+imperfection_case.InitialSway(undefined, [lc5.GetNo()], [[1.0], [5.0, 3.5, "Only one comment"], [6.0, -2.5]], 2, "USER_DEFINED_IN_V", "DIRECTION_W");
+imperfection_case.StaticDeformation(undefined, [lc6.GetNo()], "LOAD_CASE", lc7.GetNo(), 1.5, "SPECIFIC_NODE", node.no, "USER_DEFINED_IN_V");
+imperfection_case.GroupOfImperfection(undefined, [lc7.GetNo()], [[lc3.GetNo(), 3.0, "Comment 1"]], false);
+var coordinate_system = new CoordinateSystem();
+coordinate_system.Offset(undefined, [1.0, 2.0, 3.0]);
+
+/****************************************************** Member imperfections *************************************************************/
+var member_imperfection = new MemberImperfection();
+member_imperfection.InitialSway(undefined, localIC, [member.GetNo()], "PRINCIPAL", "V_NEGATIVE");
+member_imperfection.Relative(1.5);
+member_imperfection.InitialSway(undefined, localIC, [member.GetNo()]);
```

Comment:
Make a new variable for this -> it is hard to read. I spent some time to decript it how it works. So I create MemberImperfection object then I use function for selection of type of initial sway.